plugins/samples/overwrite_header/plugin.cc

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_overwrite_header]
+#include "proxy_wasm_intrinsics.h"
+
+class MyHttpContext : public Context {
+ public:
+  explicit MyHttpContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t headers,
+                                       bool end_of_stream) override {
+    // Change the key and value according to your needs
```

Comment:
Please point out in the comment how this differs from addRequestHeader (I had to go look it up). The difference is that addRequestHeader will append a value if one already exists (e.g. key: value -> key: value, value2) whereas this operation replaces the value entirely (key: value2).

You might want to show that difference in behavior by calling addResponseHeader on the response path below.

Reply:
You've updated both, but I was suggesting that one handler (e.g. request) does the existence check, whereas the other (e.g. response) does a blind replace. That way users can see the difference.

Reply:
I see... changing now

Reply:
Done

Reply:
Thanks!